lib/octokit/client/stats.rb

```diff
@@ -75,11 +75,32 @@ def punch_card_stats(repo, options = {})
       #
       # @param repo [Integer, String, Hash, Repository] A GitHub repository
       # @param metric [String] The metrics you are looking for
-      # @return [Array<Sawyer::Resource>] Magical unicorn stats
+      # @return [Array<Sawyer::Resource> or nil] Stats in metric-specific format, or nil if not yet calculated.
+      # @see https://developer.github.com/v3/repos/statistics/
       def get_stats(repo, metric, options = {})
-        data = get("#{Repository.path repo}/stats/#{metric}", options)
+        path = "#{Repository.path repo}/stats/#{metric}"
+        if retry_timeout = options.delete(:retry_timeout)
+          get_stats_with_retry(retry_timeout, path, options)
+        else
+          get(path, options)
+        end.tap do
+          return nil if last_response.status == 202
+        end
+      end
+
+      # @private Get stats for a repository. Retry call until stats are ready or retry_timeout exceeded.
+      #
+      # @param retry_timeout [String] For how long Octokit should keep trying to get stats (in milliseconds)
+      # @return [Array<Sawyer::Resource> or nil] Stats in metric-specific format, or nil if not yet calculated.
+      def get_stats_with_retry(retry_timeout, path, options = {})
```

Comment:
I'd be proud & happy, but I'm a little afraid that I'm not able to write a proper documentation for such generic method.